services/motion/client.go

```diff
@@ -97,6 +101,66 @@ func (c *client) MoveOnMap(
 	return resp.Success, nil
 }
 
+func (c *client) MoveOnGlobe(
+	ctx context.Context,
+	componentName resource.Name,
+	destination *geo.Point,
+	heading float64,
+	movementSensorName resource.Name,
+	obstacles []*referenceframe.GeoObstacle,
+	linearVelocity float64,
+	angularVelocity float64,
+	extra map[string]interface{},
+) (bool, error) {
+	ext, err := vprotoutils.StructToStructPb(extra)
+	if err != nil {
+		return false, err
+	}
+
+	if destination == nil {
+		return false, errors.New("Must provide a destination")
+	}
+
+	req := &pb.MoveOnGlobeRequest{
+		Name:               c.name,
+		ComponentName:      protoutils.ResourceNameToProto(componentName),
+		Destination:        &commonpb.GeoPoint{Latitude: destination.Lat(), Longitude: destination.Lng()},
+		MovementSensorName: protoutils.ResourceNameToProto(movementSensorName),
+		Extra:              ext,
+	}
+
+	// Optionals
+	if !math.IsNaN(heading) {
+		req.Heading = &heading
+	}
+	if len(obstacles) > 0 {
+		obstaclesProto := make([]*commonpb.GeoObstacle, 0, len(obstacles))
+		for _, eachObst := range obstacles {
+			convObst, err := referenceframe.GeoObstacleToProtobuf(eachObst)
+			if err != nil {
+				return false, err
+			}
+			obstaclesProto = append(obstaclesProto, convObst)
+		}
+		req.Obstacles = obstaclesProto
+	}
+	reqLinear := float32(linearVelocity)
+	if !math.IsNaN(linearVelocity) {
```

Comment:
[q] should linearVelocity always be positive?

Reply:
Definitely not. While the current simplified implementation has no reason to back up, that will not hold as soon as we model the robot motion properly.